components/function-controller/internal/git/client.go

```diff
@@ -0,0 +1,29 @@
+package git
+
+import (
+	gogit "github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	"github.com/go-git/go-git/v5/storage/memory"
+)
+
+type client struct{}
+
+func newClient() *client {
+	return &client{}
+}
```

Comment:
hm.. as it is internal and `client` struct has no fields then this constructor is not needed :)